hashednode.go

```diff
@@ -31,14 +31,37 @@ import (
 )
 
 type HashedNode struct {
-	commitment  []byte
+	commitment  SerializedPointCompressed
+	serialized  SerializedPoint
 	cachedPoint *Point
 }
 
+func NewHashedNode(buf SerializedPoint) *HashedNode {
+	var h HashedNode
+	h.serialized = buf
+
+	return &h
+}
+
 func (*HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
 	return errInsertIntoHash
 }
 
+func (h *HashedNode) BytesUncompressed() []byte {
+	return h.serialized
+}
```

Comment:
New method that it's used to have the serialized version of this _HashedNode_. This is trivial in this case, since our _HashedNode_ representation is literally holding the serialized bytes we got from the DB, so we simply return that.